### ISSUE TITLE
v1.5.6.7: Occasional crashes and a proposed fix  

### ISSUE THREAD

**Issue by Yaron10** [jsleroy/compare-plugin#5]
STR:
Uncheck "Ignore Spacing" and compare the files in Test.zip.

Result: 
The Plugin Manager throws an exception and Compare crashes.

Proposed fix:
Changing
~~~
#define VARRAY_INIT_SIZE 16
~~~~~~~~~
to
~~~~~~~~
 #define VARRAY_INIT_SIZE 5
~~~~~~~~~~
(in varray.h) seems to solve the problem.

[Test.zip](https://github.com/jsleroy/compare-plugin/files/93611/Test.zip)

**Comment by Yaron10** [comment]
The following (start of) line appears twice; they're the cause of the crash.
~~~~~~
added = setDiffLines
~~~~~~~

**Comment by pnedev** [comment]
Hi @Yaron10 ,

It is not clear to me if the VARRAY_INIT_SIZE causes the crash or the "added = setDiffLines" you mention in the second post.
Could you explain a bit?
Thanks! :)

BR

**Comment by Yaron10** [comment]
Hi Pavel,

Thank you for looking into this. Great! :)

I don't know if this is relevant: I didn't use VS Debugger but rather inserted some MessageBox(es) along the way.

The crash occurs when one of the "added = setDiffLines" lines is being executed.
Comparing the code in Compare.cpp to that of v.1.5.6.6, I couldn't find any possible cause.
And thus, I concluded the culprit must be somewhere else.
After some more comparisons and tests, I found that reverting "VARRAY_INIT_SIZE 16" to "VARRAY_INIT_SIZE 5" (the value in previous versions) solved the problem.

UFO is smart and he certainly mastered the code. So I believe he changed that value for a good reason. :)

Another interesting and possibly related addition (by UFO; in Compare.cpp - a few lines above "added = setDiffLines"; the same loop was added for doc2Changes too):
~~~~
	for (int i = 0; i < doc1Changed; i++)
	{
		doc1Changes[i] = {};
	}
~~~~~ 
I omitted those loops in the build I use.

If you think that reverting those changes is good enough, I'll trust your judgment.

Thanks again and best regards.
